docs/rfcs/bun-build-logs.ts

```diff
@@ -0,0 +1,40 @@
+export const enum MessageLevel {
+  err = 1,
+  warn = 2,
+  note = 3,
+  info = 4,
+  debug = 5,
+}
+
+export interface Location {
+  file: string;
+  namespace: string;
+  line: number;
+  column: number;
+  line_text: string;
+  suggestion: string;
+  offset: number;
+}
+
+export interface MessageData {
+  text?: string;
+  location?: Location;
+}
+
+export interface MessageMeta {
+  resolve?: string;
+  build?: boolean;
+}
+
+export interface Message {
+  level: MessageLevel;
+  data: MessageData;
+  notes: MessageData[];
+  on: MessageMeta;
+}
+
+export interface Log {
```

Comment:
I feel like this has an extra layer of nesting that doesn't add a lot of value.

Current:
```ts
{
   log: {
      warnings: 0,
      errors: 0,
      messages: []
   }
}
```

Alternative (simplified)
```ts
{
   errors: [],
   warnings: []
}
```